assert/assert_is_error.ts

```diff
@@ -22,18 +22,18 @@ export function assertIsError<E extends Error = Error>(
     );
   }
   if (ErrorClass && !(error instanceof ErrorClass)) {
-    msg = `Expected error to be instance of "${ErrorClass.name}", but was "${
-      typeof error === "object" ? error?.constructor?.name : "[not an object]"
-    }"${msgSuffix}`;
+    msg = `Expected error to be instance of ${JSON.stringify(ErrorClass.name)}, but was ${
+      typeof error === "object" ? JSON.stringify(error?.constructor?.name) : "[not an object]"
```

Comment:
I know Debatebly the error.name shouldn't need JSON.stringify, but runtime performance is not an issue and this will handle really janky cases (e.g. I can make ErrorClass.name contain quotes cause this is JavaScript we're talking about)

Reply:
The below script throws with `TypeError: Cannot assign to read only property 'name' of function 'class ErrorClass {}'`
```js
class ErrorClass {}
ErrorClass.name = 'foo"bar'
```
I don't think it's possible to make Class.name contain quotes.